src/commands/Mafia/nightmessages.js

```diff
@@ -0,0 +1,186 @@
+                   }
+                   else {
+                       collector2.stop();
+                   }
+               });
+               tmp_msg.react(gui_elems[0].emoji).catch(console.log); // initial reaction to start the chain reaction
+           });
+    }
+
+    // send legend & send + cancel button & create reaction collector + store it for later use as above
+    msg.channel.send(gui.legend_text())
+       .then(tmp_msg => {
+           // this somehow fails to make them appear in always the same order...
+           tmp_msg.react(gui.send.emoji).then(tmp_msg.react(gui.abort.emoji)).catch(console.log);
+           let collector = tmp_msg.createReactionCollector((_, user) => [msg.author.id].concat(bot.mafia.getModsID()).indexOf(user.id) > -1, {time: 1800000});
+           msgs.push(['send/cancel', collector]);
+           collector.on('collect', reaction => _reactionEventMessageSendCancel(reaction, bot));
+           collector.on('end', reaction => {
+               tmp_msg.clearReactions();
+               if (reaction.message !== undefined && reaction.message.reactions.array().length === 0) {
+                   tmp_msg.edit('\n\n**Menu timed out... are 30 minutes really not enough? Or did you just forgot to send/cancel?**');
+               }
+               else {
+                   tmp_msg.edit('\n\n**Finished!**');
+               }
+           });
+       });
+};
+
+// fires every time a reaction (emoji) from the command-initiator is added to one of the messages
+function _reactionEventMessageGroup(reaction, own_msg, bot) {
+    let msg = reaction.message;
+    if (reaction.emoji.name === gui.reset.emoji) {
+        msg.edit(msg.content.split('\n')[0]);
+        // This can be used to reset the gui after pressing the reset emoji but it is more irritating than usefull imo...
+        // for (let tmp_user of reaction.users.array()) {
+        //     if (tmp_user === msg.author) continue;
+        //     for (let tmp_reaction of msg.reactions.array())
+        //         tmp_reaction.remove(tmp_user);
+        // }
+        return;
+    }
+    // Handling of custom messages - only accepts messages from the one who initiated the command
+    if (reaction.emoji.name === gui.custom.emoji) {
+        msg.channel.send('\n\n**Enter your custom message for **' + msg.mentions.users.array()[0])
+           .then(send_msg => {
+               // waits for the next message (timeout 5 min) of the author and add the content of it to this persons 'to send messages' list
+               msg.channel.awaitMessages(m => [own_msg.author.id].concat(bot.mafia.getModsID()).indexOf(m.author.id) > -1, {max: 1, time: 300000})
+                  .then(collected => {
+                      msg.edit(msg.content + '\n' + collected.array()[0]);
+                      send_msg.delete();
+                      collected.array()[0].delete(); // dont know what permissions the bot will have so... maybe just comment that out
```

Comment:
Presumably this can/should be run in PM? If so, the bot can never delete anothers message. If this needs to be run in a channel we can just make it a required permission?